src/digest/sha1.rs

```diff
@@ -33,6 +33,16 @@ type W32 = Wrapping<u32>;
 /// This implementation therefore favors size and simplicity over speed.
 /// Unlike SHA-256, SHA-384, and SHA-512,
 /// there is no assembly language implementation.
+#[cfg(feature="no_asm")]
+pub unsafe fn block_data_order(state: &mut [u64; MAX_CHAINING_LEN / 8],
+                                      data: *const u8,
+                                      num: c::size_t) {
+    let data = data as *const [u8; BLOCK_LEN];
+    let blocks = core::slice::from_raw_parts(data, num);
+    block_data_order_safe(state, blocks)
```

Comment:
What I mean by my comment above is that we should make it so that in the `no_asm` mode, `block_data_order_safe` is used as the `block_data_order` function. But, I can see now that that requires additional effort, so we can defer that to a later commit.